types/src/validator_verifier.rs

```diff
@@ -159,54 +168,114 @@ impl ValidatorVerifier {
         }
     }
 
-    /// This function will successfully return when at least quorum_size signatures of known authors
-    /// are successfully verified. Also, an aggregated signature is considered invalid if any of the
-    /// attached signatures is invalid or it does not correspond to a known author. The latter is to
-    /// prevent malicious users from adding arbitrary content to the signature payload that would go
-    /// unnoticed.
-    pub fn verify_aggregated_struct_signature<T: CryptoHash + Serialize>(
+    // Generates a multi signature from partial signatures without actually verifying it.
+    pub fn generate_multi_signature(
         &self,
+        partial_signatures: &PartialSignatures,
+    ) -> Result<(MultiSignature, Option<PublicKey>), VerifyError> {
+        if partial_signatures.is_empty() {
```

Comment:
is it some tests expect it? I'd like to move it out from this function and make it return (MultiSig, PubKey) instead of Option

Reply:
Yes, there are some tests that fail without this check - let me see what tests are depending on it and how we can fix them.